src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs

```diff
@@ -843,6 +827,38 @@ private void DoSimultaneousReads(Func<Task<string>> read, string expectedValue)
             Assert.Equal(new List<Exception>(), exceptions);
         }
 
+        private void DoSimultaneousWrites(Func<string, Task> write)
+        {
+            var barrier = new Barrier(NumThreads);
+            var countdown = new CountdownEvent(NumThreads);
+
+            var exceptions = new List<Exception>();
+            for (var i = 0; i < NumThreads; i++)
+            {
+                ThreadPool.QueueUserWorkItem(s =>
+                {
+                    var id = (int)s;
+                    barrier.SignalAndWait();
+                    try
+                    {
+                        write(id + "").Wait();
+                    }
+                    catch (Exception ex)
+                    {
+                        lock (exceptions)
+                        {
+                            exceptions.Add(ex);
+                        }
+                    }
```

Comment:
logic was updated to match DoSimultaneousReads.  we need the test to not hang even if a read/write throws.  instead, we still finish all the read/writes, but report at the end if any exceptions wer encountered and we fail the test).